enhancements/agent-installer/host-based-mirror.md

```diff
@@ -0,0 +1,282 @@
+of the local registry must also be synchronized with the releaseImage provided
+in the manifests files; whenever there is an update to the releaseImage the
+local registry must also be updated.  Finally the user must add an
+ImageContentSources section to the install-config.yaml to define the registry
+sources and mirrors. This is error-prone and often results in misconfiguration
+and failed installations.
+
+This enhancement would remove the need for an additional server and
+reduce the burden on the user when using the agent based installer in a
+disconnected environment.
+
+### User Stories
+
+- As a user creating an initial Openshift cluster in a disconnected
+  environment I do not want to create and manage a registry to provide
+  images for the cluster. As I may not have access to an an external server
+  to run a registry, I need to boot the hosts for the initial cluster with no
+  additional servers.
+- As a user with limited Openshift experience I want to use an ISO that
+  fully contains all that is needed to create a cluster.
+
+### Goals
+
+- Remove the need to set up and maintain a local registry in fully-disconnected
+  environments.
+- Include in the ISO the container images and tools required for mirroring.
+- Provide an additional command to build a container image tarfile using
+  the embedded releaseImage in a connected environment.
+- When building the ISO in a disconnected environment, allow for a tarfile,
+  either from the previous item, or built using `oc-mirror`, to be included.
+- Work with all topologies supported by the agent based installer - SNO,
+  Compact, HA.
+
+### Non-Goals
+
+- Replace the current method of using a local registry on a separate server
+  in a disconnected environment, as that method will still be supported.
+- Replace the current method in a connected environment where no local
+  registry is required, that will still be supported.
+- Reduce install times (since network access is greatly reduced it is expected
+  to reduce install times, but it is not a goal of this feature).
+- Fully support upgrades and updates, as this host-based mirror is intended
+  for Day 1 only.
+
+## Proposal
+
+### Workflow Description
+
+This change affects multiple functional areas of the agent installer - the
+user interface, image (ISO) creation, and host boot. Its not feasible to
+stage these modifications over multiple Openshift releases as they are all
+required for the host registry implementation.
+
+Currently, to perform a disconnected installation, a user must add the
+ImageContentSources section to install-config.yaml with the upstream sources
+and corresponding mirror locations. With this proposal, this step is no longer
+required, instead a tarfile containing the image sources must be included in
+the assets directory when running `openshift-install agent create image`.
+A new command `openshift-install agent create container-image-file` can be
```

Comment:
Why a new command? Wouldn't be easier to add a flag to the existing one,ie `openshift-install agent create image --embed-release-image=true`?

Reply:
Good idea, changed.